### PR TITLE
Fix issues with specifying scalaHome/scalaInstance and running tests

### PR DESCRIPTION
**Pull request by jsuereth**
Fixes an issue where the classpath filter was blocking loading Scala classes inside test-runners, thanks to an invalid classpath list.

cc> @paulp Any chance you've seen this issue?  I just discovered it the other day, but given your usage of sbt, I think you may be the only one I know who would have hit this, may be able to validate it outside the Akka build/test case.

Review by @cunei cc> @eed3si9n @rkuhn

### CODE REVIEW

**Comment by paulp** [comment]
Don't know. With sbt I almost never know if something is a bug or my inability to understand how it's supposed to work. You're right this is the sort of thing I would likely have seen, but if I can trial-and-error my way to something which does what I want I discard the failed paths.

File: sbt/src/sbt-test/tests/scala-instance-classloader/src/test/scala/Test.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+class BadTest {
+
+  @Test
+  def testCpIssue(): Unit = {
+    // TODO - This is merely the laziest way to run the test.  What we want to do:
+    // * Load something from our own classloader that's INSIDE the scala library
+    // * Try to load that same something from the THREAD CONTEXT classloader.
+    // * Ensure we can do both, i.e. the second used to be filtered and broken.
+    val current = Thread.currentThread.getContextClassLoader
+    val mine = this.getClass.getClassLoader
```
Comment (rkuhn) [inline sbt/src/sbt-test/tests/scala-instance-classloader/src/test/scala/Test.scala:14]: I guess these two are left-overs from debugging? (and the comment also seems to be outdated)

**Comment by rkuhn** [comment]
LGTM (but I am not an sbt expert); in any case I applaud the `toString` methods on the classloaders.

File: sbt/src/sbt-test/tests/scala-instance-classloader/src/test/scala/Test.scala (hunk 1 above)
Comment (jsuereth) [inline sbt/src/sbt-test/tests/scala-instance-classloader/src/test/scala/Test.scala:14]: Basically.   Eventually I want to NOT use akka for this test, which is why the comment is there.  "Yours" vs. "Mine" was to try to make that happen directly (i.e. without akka).

**Comment by cunei** [comment]
The definition of `isManagedVersion` is a a bit fragile, but within the current restrictions LGTM. Cheers!

**Comment by jsuereth** [comment]
Thanks @cunei !

**Pull request merged**